catalyst/core/state.py

```diff
@@ -347,10 +356,13 @@ def __init__(
 
         self.loader_name: str = None
         self.loader_step: int = 0
+        self.loader_samples: int = 0
         self.loader_len: int = 0
+        self.loader_batch_size = 0
 
         self.batch_size: int = 0
 
+        self.global_samples: int = 0
         self.global_step: int = 0
         self.global_epoch: int = 1
```

Comment:
what do you think about naming change?
for example:
```
global_epoch - epoch counter
global_batch_step - batch counter (not sure if we need it)
global_sample_step - samples counter (should be used for logging)

loader_batch_step - current loader batch step
loader_sample_step - current loader sample step
```

Reply:
I agree with any consistent naming here. =)

About global_batch_step -- I would like to make a selectable option for loggers, to log over samples or over batches. Also, IMHO it's better to have a consistent API of `state`. For future usage or for users.